virtual_workstation_setup.sh

```diff
@@ -1,20 +1,22 @@
 # Operations in this file should be idempotent-ish. At the very least, make
 # sure your code doesn't do duplicate work/clobber existing files on rerun
-workdir=$1
-
-# TODO: Support forwarded SSH keys
-# if [[ ! -f ~/.ssh/id_rsa ]]; then
-#     echo "Please ensure your ssh keys are set up in ~/.ssh/id_rsa and ~/.ssh/id_rsa.pub; see the onboarding wiki page for more info"
-#     exit 1
-# fi
+set -euo pipefail
+workdir=${1:-$HOME}
 
 if [[ -z $(git config --get user.name) ]]; then
     echo "Please ensure your git credentials are set up; see the onboarding wiki page for more info"
     exit 1
 fi
 
-if [[ -z "$1" ]]; then
-  workdir=$HOME
+# SSH into GitHub and check for the success message. The SSH command
+# returns 1, so it can't be used alone
+github_test=$(ssh -o ConnectTimeout=5 -o StrictHostKeyChecking=no -T git@github.com 2>&1 || true)
+if ! (echo "${github_test}" | grep -q "You've successfully authenticated, but GitHub does not provide shell access"); then
```

Comment:
(Not familiar with bash if-statements by any means) but I think we generally want to double square brackets so bash test operators can be used. Consider keeping the `[[]]` for consistency

```
if [[ ! $(echo "$github_test" | grep -q "You've successfully authenticated, but GitHub does not provide shell access") ]]; then
```